test/src/main/java/org/apache/accumulo/test/SampleIT.java

```diff
@@ -0,0 +1,497 @@
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.WrappingIterator;
+import org.apache.accumulo.core.sample.RowSampler;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
+public class SampleIT extends AccumuloClusterHarness {
+
+  private static final Map<String,String> OPTIONS_1 = ImmutableMap.of("hasher", "murmur3_32", "modulus", "1009");
+  private static final Map<String,String> OPTIONS_2 = ImmutableMap.of("hasher", "murmur3_32", "modulus", "997");
+
+  private static final SamplerConfiguration SC1 = new SamplerConfiguration(RowSampler.class.getName()).setOptions(OPTIONS_1);
+  private static final SamplerConfiguration SC2 = new SamplerConfiguration(RowSampler.class.getName()).setOptions(OPTIONS_2);
+
+  public static class IteratorThatUsesSample extends WrappingIterator {
+    private SortedKeyValueIterator<Key,Value> sampleDC;
+    private boolean hasTop;
+
+    @Override
+    public boolean hasTop() {
+      return hasTop && super.hasTop();
+    }
+
+    @Override
+    public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean inclusive) throws IOException {
+
+      int sampleCount = 0;
+      sampleDC.seek(range, columnFamilies, inclusive);
+
+      while (sampleDC.hasTop()) {
+        sampleCount++;
+        sampleDC.next();
+      }
+
+      if (sampleCount < 10) {
+        hasTop = true;
+        super.seek(range, columnFamilies, inclusive);
+      } else {
+        // its too much data
+        hasTop = false;
+      }
+    }
+
+    @Override
+    public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
+      super.init(source, options, env);
+
+      IteratorEnvironment sampleEnv = env.newIEWithSamplingEnabled();
+
+      sampleDC = source.deepCopy(sampleEnv);
+    }
+  }
+
+  @Test
+  public void testBasic() throws Exception {
+
+    Connector conn = getConnector();
+    String tableName = getUniqueNames(1)[0];
+    String clone = tableName + "_clone";
+
+    conn.tableOperations().create(tableName, new NewTableConfiguration().enableSampling(SC1));
+
+    BatchWriter bw = conn.createBatchWriter(tableName, new BatchWriterConfig());
+
+    TreeMap<Key,Value> expected = new TreeMap<Key,Value>();
+    String someRow = writeData(bw, SC1, expected);
+
+    Scanner scanner = conn.createScanner(tableName, Authorizations.EMPTY);
```

Comment:
Seeing this block of code makes me think more that we should have a ScannerConfig to reduce the amount method calls over and over again. Or, some basic scan information stored in the connector itself (e.g. always enable sampling).